### PR TITLE
Support for GNS and Improved Flexibility of Time Parsing Code

### PR DESCRIPTION
**Pull request by drewf7**
Hello, I was looking into using this library for a project, and found it filled all our needs with the exception of GNS parsing, so I went ahead and added support for that.

In the process of doing so I noticed that date stamps were getting defaulted to the 19th century. I believe the desired default behavior is probably to use the current date and then UTC time stamp from the packet, so I converted the parseTime method to the one from https://github.com/infusion/GPS.js

I then added tests, and fixed the tests that I broke.

### CODE REVIEW

File: helpers.ts
Comment (101100) [inline helpers.ts]: Since the result of this method will be joined with commas, this should be:

```suggestion
        return "";
```

File: helpers.ts
Comment (101100) [inline helpers.ts]: Here too:

```suggestion
        return "";
```

File: package-lock.json
Comment (101100) [inline package-lock.json]: Please remove this file as I prefer to use `yarn`.

**Review by 101100 — CHANGES_REQUESTED** [review]
Thanks for the improved date/time parsing and the new packet type!

Just a few minor requests.

In addition, can you update `.travis.yml` to have 10, 12, 13, and 14 so the tests will pass.

**Comment by drewf7** [comment]
Thanks for the feedback!

I think I addressed everything. Let me know if that's not the case.

**Comment by drewf7** [comment]
It looks like travis is still having problems with node 4 and 14. 14 Just doesn't appear to be available at this point, but I'm not sure what's causing the 4 failure.

It looks like maybe something to do with the yarn cli?

Let me know if you'd like me to remove one or both of those to return the builds to passing. Looks good on all other versions.

**Comment by 101100** [comment]
@drewf7 Ah, looks like 14 isn't available.  I misread the chart; it is still pending.  Can you remove it and 4 and 6 (which is also EOL)?  Also, since we are only testing on v8 and above can you update `package.json` and set `engines` to:

```
  "engines": {
    "node": ">= 8.0.0"
  },
```

**Review by 101100 — APPROVED** [review]
Looks good; go ahead and merge it. 

Once this is merged I will update the dev dependencies to the latest version and release everything as version 3 (since the supported node.js version changed).

**Comment by drewf7** [comment]
@101100 I don't have a merge button available to me (lack of write access) so the honor is all yours :p

Thanks for taking the time to get this merged and released.

**Comment by 101100** [comment]
@drewf7 Version 3.0.0 has been published with your changes.  Please let me know if you have any troubles with the published version.

**Pull request merged**
